java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -1100,7 +1102,13 @@ private ConstrainedTerm applySpecRules(ConstrainedTerm constrainedTerm, List<Rul
                 global.stateLog.log(StateLog.LogEvent.SRULEATTEMPT, specRule.toKRewrite(), constrainedTerm.term(), constrainedTerm.constraint());
                 ConstrainedTerm result = buildResult(specRule, constraint, null, true, constrainedTerm.termContext(),
                         new FormulaContext(FormulaContext.Kind.SpecConstr, specRule, global));
-                global.stateLog.log(StateLog.LogEvent.SRULE, specRule.toKRewrite(), constrainedTerm.term(), constrainedTerm.constraint(), result.term(), result.constraint());
+                if (result != null) {
+                    global.stateLog.log(StateLog.LogEvent.SRULE, specRule.toKRewrite(), constrainedTerm.term(),
+                            constrainedTerm.constraint(), result.term(), result.constraint());
+                } else {
+                    global.stateLog.log(StateLog.LogEvent.SRULE, specRule.toKRewrite(), constrainedTerm.term(),
+                            constrainedTerm.constraint());
+                }
```

Comment:
We shouldn't be logging that a spec rule was applied `SRULE` if it doesn't apply correctly. So I think you can get rid of the `else` branch.

Reply:
Done.